scripts/bootstrap.sh

```diff
@@ -0,0 +1,189 @@
+    iamcredentials.googleapis.com \
+    cloudresourcemanager.googleapis.com \
+    compute.googleapis.com \
+    serviceusage.googleapis.com \
+    appengine.googleapis.com \
+    cloudbuild.googleapis.com > /dev/null
+
+# Create terraform service account
+if gcloud iam service-accounts describe \
+    $EMBLEM_TF_SERVICE_ACCOUNT \
+    --project $OPS_PROJECT &> /dev/null ; then
+        echo -e "\n\xe2\x88\xb4 Using existing Emblem Terraform service account:  $EMBLEM_TF_SERVICE_ACCOUNT "
+else
+    echo -e "\n\xe2\x88\xb4 Creating Emblem Terraform service account: $EMBLEM_TF_SERVICE_ACCOUNT "
+    gcloud iam service-accounts create emblem-terraformer \
+        --project="$OPS_PROJECT" \
+        --description="Service account for deploying resources via Terraform" \
+        --display-name="Emblem Terraformer"
+fi
+
+# Give cloud build service account token creator on terraform service account policy
+echo -e "\n\xe2\x88\xb4 Updating Terraform service account IAM policy... "
+gcloud iam service-accounts add-iam-policy-binding --project=$OPS_PROJECT \
+    $EMBLEM_TF_SERVICE_ACCOUNT \
+    --member="serviceAccount:${BUILD_SERVICE_ACCOUNT}" \
+    --role="roles/iam.serviceAccountTokenCreator" &> /dev/null
+
+# Ops permissions
+echo -e "\n\xe2\x88\xb4 Updating ops project IAM policy... "
+
+OPS_CURRENT_IAM=$(gcloud projects get-iam-policy $OPS_PROJECT --format=yaml | tail -n +2)
+
+echo -e "${OPS_IAM}\n${OPS_CURRENT_IAM}" | \
+    gcloud projects set-iam-policy $OPS_PROJECT /dev/stdin > /dev/null
+
+# App permissions for stage and prod
+
+echo -e "\n\xe2\x88\xb4 Updating stage project IAM policy... "
+
+STAGE_CURRENT_IAM=$(gcloud projects get-iam-policy $STAGE_PROJECT --format=yaml | tail -n +2)
+
+echo -e "${APP_IAM}\n${STAGE_CURRENT_IAM}" | \
+    gcloud projects set-iam-policy $STAGE_PROJECT /dev/stdin > /dev/null
+
+echo -e "\n\xe2\x88\xb4 Updating prod project IAM policy... "
+PROD_CURRENT_IAM=$(gcloud projects get-iam-policy $PROD_PROJECT --format=yaml | tail -n +2)
+echo -e "${APP_IAM}\n${PROD_CURRENT_IAM}" | \
+    gcloud projects set-iam-policy $PROD_PROJECT /dev/stdin > /dev/null
+
+# Setup Terraform state bucket
+
+if gcloud storage buckets list gs://$STATE_GCS_BUCKET_NAME &> /dev/null ; then
+    echo -e "\n\xe2\x88\xb4 Using existing Terraform remote state bucket: gs://${STATE_GCS_BUCKET_NAME} "
+    gcloud storage buckets update gs://$STATE_GCS_BUCKET_NAME --versioning > /dev/null
+else
+    echo -e "\n\xe2\x88\xb4 Creating Terraform remote state bucket: gs://${STATE_GCS_BUCKET_NAME} "
+    gcloud storage buckets create gs://${STATE_GCS_BUCKET_NAME} --project=$OPS_PROJECT > /dev/null
+    echo -e "\n\xe2\x88\xb4 Enabling versioning... "
+    gcloud storage buckets update gs://$STATE_GCS_BUCKET_NAME --versioning > /dev/null
+fi
+
+echo -e "\n\xe2\x88\xb4 Setting storage bucket IAM policy for Terraform service account..."
+gcloud storage buckets add-iam-policy-binding gs://$STATE_GCS_BUCKET_NAME \
```

Comment:
Getting an error around `add-iam-policy-binding` in my cloud shell:

<img width="842" alt="Screen Shot 2022-12-13 at 3 27 07 PM" src="https://user-images.githubusercontent.com/1331216/207467813-52bdfe96-6848-4d0b-9d4b-887da3e58ec5.png">

Reply:
Thanks for catching this!

`Google Cloud SDK 410.0.0` (default SDK in Cloud Shell) does not support `gcloud storage buckets add-iam-policy-binding`

Changed this line to use classic gsutil.